### ISSUE TITLE
Missing focus style for the 'Choose a pattern' modal dialog listbox options

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#49333]
### Description

Keyboard users need a visible focus style to see where focus is and consequently be able to use an user interface. This is a basic requirement for accessibility and usability.

When creating a new template in the site editor, the preview options in the 'Choose a pattern' modal dialog do have a _hover_ style:

<img width="971" alt="Screenshot 2023-03-24 at 10 44 09" src="https://user-images.githubusercontent.com/1682452/227502448-497c17bd-2cd3-41ea-b5da-51123109229a.png">

but they don't have a focus style:

![keyboard](https://user-images.githubusercontent.com/1682452/227502517-6ce5fd6c-e20d-4bcc-b571-f6a701134e05.gif)

Turns out in the related stylesheet there's a `:focus` rule but it's applied to an element that is not focusable....

### Step-by-step reproduction instructions

- Go to the site editor.
- Create a new _custom_ template.
- Wait.
- The 'Choose a pattern' modal dialog opens.
- Hover your mouse on the available patterns previews and observe there's a hover style.
- Use the Tab key to navigate through the focusable elements within the modal and observe the patterns previews don't have a focus style.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
